substrate/frame/multisig/src/weights.rs

```diff
@@ -46,9 +46,8 @@
 #![allow(unused_imports)]
 #![allow(missing_docs)]
 
-use frame_support::{traits::Get, weights::{Weight, constants::RocksDbWeight}};
-use core::marker::PhantomData;
-
+// TODO update this in frame-weight-template.hbs
+use frame::weights_prelude::*;
```

Comment:
Huh honestly not sure what to best do here. The long-term solution is to have a project config file for the benchmarker, like for Zepter, so that it gets all the flags from there.  
Then we could add an argument to use the `frame` crate instead. 

Until then we either have to add a `--use-frame-crate` flag manually, or cheat by re-exporting the weight type from the crate and use that.

Reply:
This would break both for `frame` and `polkadot-sdk`, basically meaning pallet development always needs `frame_support` in place? bit unfortunate. 

How about providing two `frame-weight-template.hbs` ? One for old school, one for new school?

Reply:
Discussed about moving all our pallets to include `frame` as dependency and moving the template to assume that exists.

Reply:
maybe we could also assume that the pallet weight is a direct module inside the module with pallet definition.

So we can use `super::pallet::reexport_of_frame`?

But assuming `frame` as a dependency sounds better, people can always modify the `.hbs` to their flavor.

Reply:
the handlebar templates should be able to allow whatever decision you all pick, but the change needs to happen at that level, not here.

Reply:
I see two ways forward with this: 

1. We currently use `substrate/.maintain/frame-weight-template.hbs` in most places. We create a new `substrate/.maintain/frame-umbrella-weight-template.hbs`. Users are free to choose. Our CI should also adapt, and based on the presence of `frame-support` in the deps, make a decision.
2. We make a mega PR that will move all pallets to have a `#[doc(hidden)] pub(crate) mod __private_weights { /* re export all the stuff that might be needed in the weight.rs file */ }`. We also update `substrate/.maintain/frame-weight-template.hbs` to start with `use crate::__private_weights::*`. Our CI should pass once we merge this. Then, we resume this PR. 

Solution 1 is clearly better, and more extensible. Solution 2 is only a reasonable hack if 1 is too hard to pull off. 

@mordamax this is mainly a question from you; how hard will it be to have our weight generation CI (the fellowship's) be able to handle two templates? or perhaps we already support it?

Reply:
I talked to @mordamax and we should explore option, it should not have any major blockers and just needs more updates to our CI.

Reply:
It is not the cleanest thing in the world, but this is how it would look like: 

2d8323f40eb51f81fb66c6a0e969454b16514b37

Reply:
Going with Solution 1: I tested your fix from https://github.com/paritytech/polkadot-sdk/commit/2d8323f40eb51f81fb66c6a0e969454b16514b37 and it works. Also changed the `run_all_benchmarks.sh` script to use both templates in https://github.com/paritytech/polkadot-sdk/pull/5995/commits/49a4cc392aef0a5e2bb5f04b39b4fb17399cf9dd. I don't see any other changes to do.

Reply:
Final CI fix in https://github.com/paritytech/polkadot-sdk/pull/5995/commits/21cd6da9b4dfece0f6e84ce99cdad3e3b518ed0a